js/modules/k6/http/http_request.go

```diff
@@ -461,25 +462,14 @@ func (h *HTTP) request(ctx context.Context, preq *parsedHTTPRequest) (*HTTPRespo
 			authorization := challenge.ToAuthorizationStr()
 			preq.req.Header.Set(digest.KEY_AUTHORIZATION, authorization)
 		}
-		trail := tracer.Done()
-
-		if state.Options.SystemTags["ip"] && trail.ConnRemoteAddr != nil {
-			if ip, _, err := net.SplitHostPort(trail.ConnRemoteAddr.String()); err == nil {
-				tags["ip"] = ip
-			}
-		}
-		trail.SaveSamples(stats.NewSampleTags(tags))
-		delete(tags, "ip")
-		stats.PushIfNotCancelled(ctx, state.Samples, trail)
 	}
 
 	if preq.auth == "ntlm" {
 		ctx = netext.WithAuth(ctx, "ntlm")
```

Comment:
To avoid the creation of a whole new context just to say "I want to use NTLM" (and also the subsequent context check `if GetAuth(req.Context()) == "ntlm" && req.URL.User != nil {...}` on every request that involves type assertions...), can't we just pass whether we want to use NTLM in the creation of the wrapper above (`transport := netext.NewTransport(...)`) - we already have `preq` there.